content/docs/command-reference/stage/list.md

```diff
@@ -10,55 +10,55 @@ usage: dvc stage list [-h] [-q | -v]
                       [targets ...]
 
 positional arguments:
-  targets          Show stages from a dvc.yaml file or a directory (with -R).
-                   'dvc.yaml' by default.
+  targets       Limit command scope to these dvc.yaml files,
+                directories (with -R), or stage names.
+                './dvc.yaml' by default.
 ```
 
+> See [`targets`](#options) for more details.
+
 ## Description
 
-`dvc stage list` is useful for quick listing of stages present in the project
-without having to open `dvc.yaml` files.
+Prints a summarized list of stages including their names and a one-line
+description (which can be omitted using `--names-only`). This command is useful
+for discovering or reviewing what stages are present in the project without
+having to examine `dvc.yaml` files manually. It can also be a shorter
+alternative to `dvc dag`.
 
-It provides a list of stages, with a stage name and the description of the
-stage. The description of the stage is read from
-[`desc` field in the `dvc.yaml` file](doc/user-guide/project-structure/pipelines-files#stage-entries),
-or if it does not exist, DVC generates the short description from the other
-metadata it has about the stage.
+Without any `targets`, `dvc stage list` lists the stages from `./dvc.yaml` (in
+the current working directory) by default.
 
-Without any targets, by default, it only lists the stages from the `dvc.yaml`
-present in the current working directory.
+The stage description is read from the `desc` field in `dvc.yaml` (truncated to
+80 characters) if it exists. Otherwise, DVC generates one with the stage's most
+important characteristics (dependencies, outputs, or metrics).
 
-DVC also uses this command to provide autocompletion in `bash` and `zsh`.
+> (ℹ️) DVC uses this command to provide
+> [shell tab completion](/doc/install/completion).
 
 ## Options
 
-- `targets` (optional command argument) - stages to list from (`./dvc.yaml` by
-  default). Different things can be provided as targets depending on the flags
-  used (more details in each option). Examples:
+- `targets` (optional command argument) - where to look for stages to list
+  (`./dvc.yaml` by default). Examples:
 
   - `dvc stage list linear/dvc.yaml`: Specific `dvc.yaml` file to list from
   - `dvc stage list -R pipelines/`: Directory path to explore recursively for
-    `dvc.yaml` files
-  - `dvc stage list train-model`: Specific stage name or a
-    [foreach group](doc/user-guide/project-structure/pipelines-files#foreach-stages)
-    in `./dvc.yaml`
-  - `dvc stage list modeling/dvc.yaml:prepare`: Stage or a
-    [foreach group](doc/user-guide/project-structure/pipelines-files#foreach-stages)
-    in a specific `dvc.yaml` file
+    stages
+  - `dvc stage list train-model`: Specific stage(s) from `./dvc.yaml`
+  - `dvc stage list modeling/dvc.yaml:prepare`: Stage(s) from a specific
```

Comment:
Again, I think it's easier to digest if it's consistent. I'm okay with having different stage names since there are different `dvc.yaml` paths, but IMO using the same stage name is probably better, too, since it's otherwise a bit hard to tell that the part after the colon is a stage name. If you prefer different stage names, I'm fine with it.

```suggestion
  - `dvc stage list linear/dvc.yaml:train-model`: Stage(s) from a specific
```

Reply:
> okay with having different stage names
but IMO using the same stage name is probably better

Did you mean different dirs but same stage name? Or vice versa 🙂 

In any case I like the variation and it should be clear that the part before `:` is a path to a dvc.yaml file.

You may be right but this seems subjective so I'd err on the side of keeping it as-is (I didn't change that in this PR). Or we can ask someone to cast a deciding vote.